### PR TITLE
rely on __future__'s unicode_literals

### PR DESCRIPTION
**Pull request by rgbkrk**
We ran into an issue where a user had some `!pip install` that had unicode output that wasn't being encoded and decoded properly. This switches the section of code that mixed string concatenation between Python 2 strings and unicode strings over to only using unicode strings. While I was at it I brought in `unicode_literals` from `__future__`.

/cc @mpacer @charsmith

### CODE REVIEW

**Comment by mpacer** [comment]
I'd also add  `# -*- coding: utf-8 -*-` to the beginning of most of those files in case the issue has to do with parsing the literal string sources when the underlying system encoding may not be utf-8.

See https://stackoverflow.com/questions/4872007/where-does-this-come-from-coding-utf-8 for a quick explanation
See https://www.python.org/dev/peps/pep-0263/ for a more comprehensive look into why this is needed.

**Comment by rgbkrk** [comment]
Ok cool, thanks. Feel free to push to this branch. We could probably add another test that exercises this as well.

**Comment by mpacer** [comment]
I'd definitely push if I were able to reproduce the error locally :/ so I can know that this makes a difference.

File: papermill/execute.py [hunk 1]
```diff
@@ -106,26 +112,31 @@ def log_outputs(cell):
     for output in cell.get("outputs", []):
         if output.output_type == "stream":
             if output.name == "stdout":
-                stdouts.append("".join(output.text))
+                stdouts.append(u"".join(output.text))
```
Comment (mpacer) [inline papermill/execute.py]: These shouldn't be necessary anymore.

Comment (rgbkrk) [reply papermill/execute.py]: Yeah since we imported Unicode literals from future

**Pull request merged**